sympy/controlsys/lti.py

```diff
@@ -0,0 +1,757 @@
+
+    def __init__(self, arg):
+
+        if isinstance(arg, TransferFunctionModel):
+
+            # call the private method for realization finding
+            self.represent = self._find_realization(arg.G, arg.s)
+
+            # create a block matrix [[A,B], [C,D]] for visual representation
+            self.BlockRepresent = BlockMatrix([[self.represent[0], self.represent[1]],
+                                               [self.represent[2], self.represent[3]]])
+            return None
+
+        else:
+            # store the argument as representation of the system
+            try:
+                self.represent = arg[:4]
+            except TypeError:
+                raise TypeError("'repesentation' must be a list-like object")
+
+            try:
+                # assert that A,B,C,D have matching shapes
+                if not ((self.represent[0].shape[0] == self.represent[1].shape[0]) and
+                        (self.represent[0].shape[1] == self.represent[2].shape[1]) and
+                        (self.represent[1].shape[1] == self.represent[3].shape[1]) and
+                        (self.represent[2].shape[0] == self.represent[3].shape[0])):
+                    raise ShapeError("Shapes of A,B,C,D must fit")
+
+                # create a block matrix [[A,B], [C,D]] for visual representation
+                self.BlockRepresent = BlockMatrix([[self.represent[0], self.represent[1]],
+                                                   [self.represent[2], self.represent[3]]])
+                return None
+
+            except TypeError:
+                raise TypeError("entries of 'representation' must be matrices")
+            except AttributeError:
+                raise TypeError("entries of 'representation' must be matrices")
+            except IndexError:
+                raise TypeError("'representation' must have at least 4 matrix-valued entries")
+
+    def _find_realization(self, G, s):
+        """ Represenatation [A, B, C, D] of the state space model
+
+        Returns the representation in state space of a given transfer function
+
+        Parameters
+        ==========
+
+        G: Matrix
+            Matrix valued transfer function G(s) in laplace space
+        s: symbol
+            variable s, where G is dependent from
+
+        See Also
+        ========
+
+        Utils : some quick tools for matrix polynomials
+
+        References
+        ==========
+
+        Joao P. Hespanha, Linear Systems Theory. 2009.
+        """
+
+        A, B, C, D = 4 * [None]
+
+        try:
+            m, k = G.shape
+
+        except AttributeError:
+            raise TypeError("G must be a matrix")
+
+        # test if G is proper
+        if not _is_proper(G, s, strict=False):
+            raise ValueError("G must be proper!")
+
+        # define D as the limit of G for s to infinity
+        D = G.limit(s, oo)
+
+        # define G_sp as the (stricly proper) difference of G and D
+        G_sp = simplify(G - D)
```

Comment:
These last two commands can potentially take a long time for complicated expressions, especially simplify. Simplify is typically an option or user commanded function due to it's potential for being slow. I would leave it out here.